java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -559,6 +563,38 @@ private static WorkspaceEdit generateTestsFromTest(ICompilationUnit unit, Compil
                 methodList, true /*pickMany*/);
     }
 
+    private static IJavaProject determineTestProject(ICompilationUnit unit) {
+        final IJavaProject javaProject = unit.getJavaProject();
+        if (TestKindProvider.getTestKindsFromCache(javaProject).size() > 0) {
+            return javaProject;
+        }
+
+        final IJavaProject[] javaProjects = ProjectUtils.getJavaProjects();
+        final List<Option> javaTestProjects = new LinkedList<>();
+        for (final IJavaProject project : javaProjects) {
+            if (project.equals(JavaLanguageServerPlugin.getProjectsManager().getDefaultProject())) {
+                continue;
+            }
+            if (TestKindProvider.getTestKindsFromCache(project).size() > 0) {
+                javaTestProjects.add(new Option(project.getHandleIdentifier(), project.getElementName(),
+                    project.getProject().getLocation().toOSString()));
+            }
+        }
+
+        if (javaTestProjects.size() == 0) {
+            JavaLanguageServerPlugin.getInstance().getClientConnection().showNotificationMessage(MessageType.Error,
+                    "The projects in the workspace do not have a test framework," +
+                    " please make sure they are configured correctly.");
```

Comment:
No test library found in your workspace, please add a test library to your project classpath first.